components/calendar/src/calendar_arithmetic.rs

```diff
@@ -125,6 +126,42 @@ pub(crate) trait CalendarArithmetic: Calendar {
     }
 }
 
+/// Trait for converting from era codes, month codes, and other fields to year/month/day ordinals.
+pub(crate) trait DateFieldsResolver: Calendar {
+    type YearInfo: PartialEq;
```

Comment:
issue: we use the name `YearInfo` for the associated type on `Calendar`, which is either, `EraYear`, `CyclicYear`, or the `YearInfo` struct. please use a different name for this, and  also document what this is

Reply:
It's the same as the `YearInfo` associated type on `CalendarArithmetic`, which is different than the `Year` associated type on `Calendar`.